Nodejs/Product/Nodejs/Intellisense/ExpansionClient.cs

```diff
@@ -0,0 +1,308 @@
+﻿//*********************************************************//
+//    Copyright (c) Microsoft. All rights reserved.
+//    
+//    Apache 2.0 License
+//    
+//    You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+//    
+//    Unless required by applicable law or agreed to in writing, software 
+//    distributed under the License is distributed on an "AS IS" BASIS, 
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
+//    implied. See the License for the specific language governing 
+//    permissions and limitations under the License.
+//
+//*********************************************************//
+using System;
+using System.Collections.Generic;
+using Microsoft.NodejsTools.Analysis;
+using Microsoft.NodejsTools.Parsing;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Editor.OptionsExtensionMethods;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudioTools;
+using MSXML;
+
+namespace Microsoft.NodejsTools.Intellisense {
+    internal sealed class ExpansionClient : IVsExpansionClient {
+        internal IServiceProvider serviceProvider;
+
+        private readonly IVsTextLines _lines;
+        private readonly IVsExpansion _expansion;
+        private readonly IVsTextView _view;
+        private readonly ITextView _textView;
+        private readonly IVsEditorAdaptersFactoryService _adapterFactory;
+        private readonly IServiceProvider _serviceProvider;
+        private IVsExpansionSession _session;
+        private bool _sessionEnded, _selectEndSpan;
+        private ITrackingPoint _selectionStart, _selectionEnd;
+
+        public const string SurroundsWith = "SurroundsWith";
+        public const string SurroundsWithStatement = "SurroundsWithStatement";

```

Comment:
You shouldn't need to use SurroundsWithStatement in JavaScript - this looks like a remnant of the python codebase. Is there a reason it's still here?